Add docs for feature Extend CSI Snapshot support to Longhorn snapshot (#455)
longhorn/website: content/docs/1.3.0/snapshots-and-backups/csi-snapshot-support/_index.md

---
title: CSI Snapshot Support
description: Creating and Restoring Longhorn Snapshots/Backups via the kubernetes CSI VolumeSnapshot mechanism
weight: 3
---

## History
- GitHub Issues:
  - https://github.com/longhorn/longhorn/issues/304
  - https://github.com/longhorn/longhorn/issues/2534
- Longhorn Enhancement Proposals:
  - https://github.com/longhorn/longhorn/blob/master/enhancements/20200904-csi-snapshot-support.md
  - https://github.com/longhorn/longhorn/blob/master/enhancements/20220110-extend-csi-snapshot-to-support-longhorn-snapshot.md
- Availability:
  - CSI VolumeSnapshot associated with Longhorn backup is available since v1.1.0
  - CSI VolumeSnapshot associated with Longhorn snapshot is available since v1.3.0

## Overview

In Kubernetes, a [CSI VolumeSnapshot](https://kubernetes.io/docs/concepts/storage/volume-snapshots/) represents a snapshot of a volume on a storage system.
When you create a CSI VolumeSnapshot, the storage system will take a snapshot of the corresponding volume.
You can then use the CSI VolumeSnapshot to create a new volume that has the same content as the content of the CSI VolumeSnapshot.

> In Longhorn, there are 2 concepts related to the content of a volume at particular moment: [Longhorn backup](../../concepts/#3-backups-and-secondary-storage) and [Longhorn snapshot](../../concepts/#24-snapshots).

Reviewer: Suggest describes snapshot first then backup.

```suggestion
In Longhorn, there are 2 concepts related to the content of a volume at a particular moment: [Longhorn snapshot](../../concepts/#24-snapshots) and [Longhorn backup](../../concepts/#3-backups-and-secondary-storage).
```

Author: Thank you. Fixed

Longhorn snapshot is the content of a Longhorn volume at a particular moment. It is stored inside the cluster.

> A Longhorn backup is associated to a Longhorn snapshot, but it is stored externally inside a [backup target](../backup-and-restore/set-backup-target/).

Reviewer: ```suggestion
A Longhorn backup is associated with a Longhorn snapshot, but it is stored externally inside a [backup target](../backup-and-restore/set-backup-target/).
```

> When you create a CSI VolumeSnapshot, you can specify whether the CSI VolumeSnapshot should be associated to a Longhorn snapshot (content is stored inside cluster) or a Longhorn backup (content is stored both inside cluster and uploaded to external backup target).

Reviewer: ```suggestion
When you create a CSI VolumeSnapshot, you can specify whether the CSI VolumeSnapshot should be associated with a Longhorn snapshot (content is stored inside cluster) or a Longhorn backup (content is stored both inside the cluster and uploaded to external backup target).
```

Author: Thank you. Fixed

> After install the [prerequisite](./enable-csi-snapshot-support), you can see [CSI VolumeSnapshot Associated with Longhorn Backup](./csi-volume-snapshot-associated-with-longhorn-backup) and [CSI VolumeSnapshot Associated with Longhorn Snapshot](./csi-volume-snapshot-associated-with-longhorn-snapshot) for more detail about how to create/restore a CSI VolumeSnapshot and the perameters you can set.

Reviewer: Suggest describes snapshot first then backup.

```suggestion
After installing the [prerequisite](./enable-csi-snapshot-support), you can see [CSI VolumeSnapshot Associated with Longhorn Snapshot](./csi-volume-snapshot-associated-with-longhorn-snapshot) and [CSI VolumeSnapshot Associated with Longhorn Backup](./csi-volume-snapshot-associated-with-longhorn-backup) for more detail about how to create/restore a CSI VolumeSnapshot and the parameters you can set.
```

Author: Thank you. Fixed
